### PR TITLE
[WIP] Added blog post and images for GraphQL Testing

### PR DESCRIPTION
**Pull request by lobut**

### CODE REVIEW

File: source/_posts/pragmatic-testing-with-graphql.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: "Pragmatic Testing with GraphQL"
+date: 2017-06-16 15:41:16
+author: rluong
+comments: true
+tags: [Testing, Engineering, Acceptance tests]
+---
+
+We’ve been using GraphQL at OpenTable for a little over half a year now.  I won’t get into detail with why we use GraphQL, sufficed to say, we really enjoyed creating our very first GraphQL endpoint.  It eased a lot of the inconsistencies that we were experiencing with some of our REST-ful services.
```
Comment (Kimtaro) [inline source/_posts/pragmatic-testing-with-graphql.md]: `I won’t go into detail as to why we started using it, but suffice to say, we really enjoyed creating our very first GraphQL endpoint.`

File: source/_posts/pragmatic-testing-with-graphql.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+The rest of this post assumes you’ve had some experience building a GraphQL endpoint.  If you aren’t familiar with it.  I want to say it feels a lot like having a querying language accessible via an HTTP endpoint.  If you want to try it for yourself, I would recommend [GitHub’s endpoint](https://developer.github.com/v4/explorer/).  To get started with GraphQL, [the official documentation](http://graphql.org/learn/) is a perfect place.
```
Comment (Kimtaro) [inline source/_posts/pragmatic-testing-with-graphql.md]: REST-ful/rest and we're at the start of the post so there is no "rest", so `This post assumes`

Comment (Kimtaro) [inline source/_posts/pragmatic-testing-with-graphql.md]: `you have some experience`

Comment (Kimtaro) [inline source/_posts/pragmatic-testing-with-graphql.md]: If / If, so `For those of you who aren’t familiar with it, it feels a lot like having a querying language accessible via an HTTP endpoint.`

Comment (Kimtaro) [inline source/_posts/pragmatic-testing-with-graphql.md]: I ~~would~~ recommend

File: source/_posts/pragmatic-testing-with-graphql.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+## The Approach
+
+So we took a step back and looked at the code we were writing.  Most of the connectors we wrote were fairly anemic.  We didn’t have much logic for our API nor did we want any.  We could have written unit tests for each parts of the schema but mocking portions of the system seemed more work than it was worth.
+
+So where did we start?  I personally enjoy the outside-in approach and started writing a few acceptance tests. Most of the errors that occurred when we extended existing schemas and inadvertently broke existing queries.  To curtail this, I wrote a query for each schema we had.  It would fire an HTTP request and expect a 200 status code and have no errors in the resulting JSON body.  Thanks to GraphQL most errors are handily reported using this mechanism so we would be able to make a decent impact using little work.
```
Comment (Kimtaro) [inline source/_posts/pragmatic-testing-with-graphql.md]: "Most of the errors that occurred when we extended existing schemas and inadvertently broke existing queries"

Start and end of this sentence don't go together, not 100% what you want to say.

File: source/_posts/pragmatic-testing-with-graphql.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+{% img left /images/posts/graphql-base-tests.gif %}
+
+As you can see, we found that this would be good enough at the time but eventually found that it would not scale very well.  If you forget to update one of these queries, you could leave out a major section out accidentally.  Being a big fan of automation, we wondered how we could scale this out.
+
+## The Solution
+
+We knew GraphiQL (the web interface for testing GraphQL queries) was using introspection to give us intellisense.  What a great idea: we could leverage introspection in a similar way to discover all the possible queries for a given endpoint.  After doing so, we repeated our previous tests which was to: fire an HTTP request and ensure there were no errors.
```
Comment (Kimtaro) [inline source/_posts/pragmatic-testing-with-graphql.md]: `previous tests, which was to fire an HTTP request `

**Comment by ajcw** [comment]
@lobut is this still WIP or can I merge and publish?

**Comment by lobut** [comment]
You can merge, but I kinda wanna see the HTML that gets generated?

**Pull request merged**